Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c

```diff
@@ -0,0 +1,1540 @@
+    MethodParam[2].Type     = AmlMethodParamTypeArg;
+    MethodParam[2].Data.Arg = 0x2; // Arg2 is the third argument to the method
+    MethodParam[3].Type     = AmlMethodParamTypeArg;
+    MethodParam[3].Data.Arg = 0x3; // Arg3 is the fourth argument to the method
+    // call the \\_SB.OSCI method
+    Status = AmlCodeGenInvokeMethod (
+               "\\_SB.OSCI",
+               4,
+               MethodParam,
+               OscMethod
+               );
+    if (EFI_ERROR (Status)) {
+      ASSERT_EFI_ERROR (Status);
+      return Status;
+    }
+  }
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_INFO,
+      "%a: Failed with Status: %r, Not Critical return SUCCESS\n",
+      __func__,
+      Status
+      ));
+    Status = EFI_SUCCESS;
+  }
+
+  return Status;
+}
+
+/**
+  Insert Pcie base size into the AML table
+
+  @param[in,out]  CrsNode    - AmlLib table node
+
+  @retval         EFI_SUCCESS, various EFI FAILUREs.
+**/
+EFI_STATUS
+EFIAPI
+InternalInsertPciExpressBaseSize (
+  IN OUT  AML_OBJECT_NODE_HANDLE  CrsNode
+  )
+{
+  EFI_STATUS  Status;
+  UINT64      RangeLen;
+  UINT64      RangeMax;
+  UINT64      RangeMin;
+
+  RangeMin = PcdGet64 (PcdPciExpressBaseAddress);
+  RangeLen = PcdGet64 (PcdPciExpressBaseSize);
+  RangeMax = RangeMin + RangeLen - 1;
+
+  Status = AmlCodeGenRdQWordMemory (
+             FALSE,
+             TRUE,
+             TRUE,
+             TRUE,
+             FALSE, // non cacheable
+             TRUE,
+             0x0,
+             RangeMin,
+             RangeMax,
+             0x0,
+             RangeLen,
+             0,
+             NULL,
+             0,
+             TRUE,
+             CrsNode,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  return EFI_SUCCESS;
+}
+
+/**
+  Locates an existing ACPI Table
+
+  @param[in]      Signature     - The Acpi table signature
+  @param[in]      OemTableId    - The Acpi table OEM Table ID. Ignored if 0
+  @param[out]     Table         - Table if Found or NULL
+
+  @retval         EFI_SUCCESS, various EFI FAILUREs.
+**/
+EFI_STATUS
+EFIAPI
+GetExistingAcpiTable (
+  IN      UINT32            Signature,
+  IN      UINT64            OemTableId,
+  OUT  EFI_ACPI_SDT_HEADER  **Table
+  )
+{
+  EFI_ACPI_SDT_HEADER     *LocalTable;
+  EFI_ACPI_TABLE_VERSION  LocalVersion;
+  EFI_STATUS              Status;
+  UINTN                   Index;
+  UINTN                   LocalTableKey;
+
+  Status = EFI_NOT_FOUND;
+  *Table = NULL;
+
+  for (Index = 0; ; Index++) {
+    Status = mAcpiSdtProtocol->GetAcpiTable (Index, &LocalTable, &LocalVersion, &LocalTableKey);
+    if (EFI_ERROR (Status)) {
+      return Status;
+    }
+
+    if (!(LocalTable->Signature == Signature)) {
+      continue;
+    }
+
+    // Accept table if OemTableId is zero.
+    if ((OemTableId == 0) ||
+        (CompareMem (&LocalTable->OemTableId, &OemTableId, 8) == 0))
+    {
+      *Table = LocalTable;
+      return EFI_SUCCESS;

```

Comment:
return should be outside of for loop?

Reply:
if we found the table, then no need to continue the loop.
Hence returns immediate.